### DISCUSSION TITLE
Example of set/get GPIO value - MQTT, HTTP, Console

### DISCUSSION THREAD

**Discussion by jfilaa** [emsesp/EMS-ESP32#1270] [General]
Can You please write few examples how to set/get value of GPIO pin via MQTT, HTTP and Console of EMS-ESP?

**Comment by jfilaa** [comment]
By HTTP:
```
curl http://192.168.2.110/api/analogsensor/setvalue \
-H 'Content-Type: application/json' \
-H 'Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ8.eyJ1c2VybmFtZSI6ImFkbWluIiwiYWRtaW4iOnRydWUsInZlcnNpb24iOiIzLjEuMWIwIn0.qeGT53Aom4rDYeIT1Pr4BSMdeWyf4_zN9ue2c51ZnM0' \
-d '{ "value":1, "id":40 }'
```
By Telnet:
`call analogsensor setvalue 0 40`

By MQTT:
`mosquitto_pub -t "ems-esp/analogsensor/" -m '{"cmd":"setvalue", "data": 0, "id": 40}'`

**Reply by proddy** [reply]
thanks @jfilaa

**Reply by proddy** [reply]
if you think we need to improve the wiki documentation at https://emsesp.github.io/docs/Commands/ let me know.

**Comment by MichaelDvP** [comment]
The `setvalue` command with gpio number is old style for compatibility, better use the unifiied api structure with entitiy name as command.
- create a analogsensor as digitalout with gpio 40 and name it for example `relais1`. Then you have:
**Telnet:** `call analogsensor relais1 on`
**Mqtt:** Topic `analogsensor/relais1`, Payload: `on` (or json `{"value":"on"}`)
**Webapi:**  `http://emsesp/api/analogsensor/relais1`, Body: `{"value":"on"}`
Value can be in all cases on/off/true/false/1/0 or translated on/off

(mqtt can also use Topic `analogsensor`, Payload `{"cmd":"relais1", "value":"on"}`)

**Reply by jfilaa** [reply]
Thank You much. If You can update wiki it would help others. By the way, **Scheduler** syntax is Command:`boiler/relais1` Value:`on`.

**Reply by proddy** [reply]
I will update the doc - need to make some changes anyway

**Reply by MichaelDvP** [reply]
@proddy some additions to the command doc in https://github.com/emsesp/docs/blob/commands_update/docs/Commands.md
@jfilaa the scheduler have same scheme, if you like to schedule the analogsensor `relais1` the command is `analogsensor/relais1`
